horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -982,4 +980,150 @@ public QueryResult queryDataSet(Integer datasetId, String jsonpath, boolean arra
       }
       return result;
    }
+
+   private void transform(Run run, String role) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role);
+      Test test = Test.findById(run.testid);
+      List<JsonNode> nodesToProcess = new ArrayList<JsonNode>();
+      boolean matched = false;
+
+      if (!runSchemas.isEmpty()){
+         for ( RunSchema s : runSchemas ){
+            if (hasMatch(s, run.data)) {
+               nodesToProcess.add(run.data);
+            }
+            int nodeSize = nodesToProcess.size();
+
+            if (nodeSize == 0) {
+               new DataSet(run.start, run.stop, "",
+                     test.id, run.data, run, 0, run.owner, run.access)
+                     .persist();
+            } else {
+               Collection<Transformer> transformers = test.transformers;
+               if (transformers.isEmpty()) {
+                  new DataSet(run.start, run.stop, "",
+                        test.id, run.data, run, 0, run.owner, run.access)
+                        .persist();
+               } else {
+                  for (Transformer t : transformers){
+                     StringBuilder array = new StringBuilder("[");
```

Comment:
The input for `Transformer.function` is not an array, it's an object keyed by the name of each extractor. That's why we have them named. E.g. if you have extractors `foo: $.x` and `bar: $.y` the input for the function can look like `{ "foo": 42, "bar": "the result of $.y" }`.
Also it would be nicer to build it as JsonNode rather than string (you'll save yourselves of dealing with commas etc.)

Reply:
ok